src/pipecat/services/aws/agent_core.py

```diff
@@ -0,0 +1,189 @@
+            aws_region: AWS region.
+            context_to_payload_transformer: Optional callable to transform LLMContext into AgentCore
+                payload dict. If None, uses default_context_to_payload_transformer.
+            response_to_text_transformer: Optional callable to extract text from AgentCore response
+                JSON. If None, uses default_response_to_text_transformer.
+            **kwargs: Additional arguments passed to parent FrameProcessor.
+        """
+        super().__init__(**kwargs)
+
+        self._agentArn = agentArn
+        self._aws_session = aioboto3.Session()
+
+        # Store AWS session parameters for creating client in async context
+        self._aws_params = {
+            "aws_access_key_id": aws_access_key or os.getenv("AWS_ACCESS_KEY_ID"),
+            "aws_secret_access_key": aws_secret_key or os.getenv("AWS_SECRET_ACCESS_KEY"),
+            "aws_session_token": aws_session_token or os.getenv("AWS_SESSION_TOKEN"),
+            "region_name": aws_region or os.getenv("AWS_REGION", "us-east-1"),
+        }
+
+        # Set transformers with defaults
+        self._context_to_payload_transformer = (
+            context_to_payload_transformer or default_context_to_payload_transformer
+        )
+        self._response_to_text_transformer = (
+            response_to_text_transformer or default_response_to_text_transformer
+        )
+
+    async def process_frame(self, frame: Frame, direction: FrameDirection):
+        """Process incoming frames and handle LLM message frames.
+
+        Args:
+            frame: The incoming frame to process.
+            direction: The direction of frame flow in the pipeline.
+        """
+        await super().process_frame(frame, direction)
+        if isinstance(frame, (LLMContextFrame, OpenAILLMContextFrame)):
+            payload = self._context_to_payload_transformer(frame.context)
+            if payload:
+                async with self._aws_session.client(
+                    "bedrock-agentcore", **self._aws_params
+                ) as client:
+                    response = await client.invoke_agent_runtime(
+                        agentRuntimeArn=self._agentArn, payload=json.dumps(payload).encode()
+                    )
+                    is_multi_part_response = "text/event-stream" in response.get("contentType", "")
+                    async for part in response.get("response", []):
+                        part_string = part.decode("utf-8")
+                        if is_multi_part_response:
+                            part_string = part_string[6:]  # omit "data: "
+                        print(f"[pk] AWS AgentCore response: {part_string}")
+                        part_json = json.loads(part_string)
+                        text = self._response_to_text_transformer(part_json)
+                        if text:
+                            await self.push_frame(LLMFullResponseStartFrame())
```

Comment:
not quite sure why i had to sandwich every single part of the response in `LLMFullResponseStartFrame`/`LLMFullResponseEndFrame` for TTS to work, but i had to...can investigate